trace-normalization/src/normalization/normalize_utils.rs

```diff
@@ -0,0 +1,191 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0. This product includes software
+// developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present
+// Datadog, Inc.
+
+use crate::errors;
+
+// DEFAULT_SPAN_NAME is the default name we assign a span if it's missing and we have no reasonable fallback
+pub const DEFAULT_SPAN_NAME: &str = "unnamed_operation";
+// DEFAULT_SERVICE_NAME is the default name we assign a service if it's missing and we have no reasonable fallback
+pub const DEFAULT_SERVICE_NAME: &str = "unnamed-service";
+
+// MAX_NAME_LEN the maximum length a name can have
+pub const MAX_NAME_LEN: usize = 100;
+// MAX_SERVICE_LEN the maximum length a service can have
+pub const MAX_SERVICE_LEN: usize = 100;
+
+pub const MAX_TAG_LEN: usize = 200;
+
+// TruncateUTF8 truncates the given string to make sure it uses less than limit bytes.
+// If the last character is an utf8 character that would be splitten, it removes it
+// entirely to make sure the resulting string is not broken.
+pub fn truncate_utf8(s: String, limit: usize) -> String {
+    if s.len() <= limit {
+        return s
+    }
+    let mut prev_index = 0;
+    for i in 0..s.len() {
+        if i > limit {
+            return s[0..prev_index].to_string();
+        }
+        prev_index = i;
+    }
+    s
+}
+
+// NormalizeService normalizes a span service and returns an error describing the reason
+// (if any) why the name was modified.
+// TODO: Implement this in a future PR
+// pub fn normalize_service(svc: String, lang: String) -> (String, Option<errors::NormalizeErrors>) {
+    // if svc == "" {
+    //     return (fallback_service(lang), errors::NormalizeErrors::ErrorEmpty);
+    // }
+    // if svc.len() > MAX_SERVICE_LEN {
+    //     return (truncate_utf8(svc, MAX_SERVICE_LEN), errors::NormalizeErrors::ErrorTooLong.into());
+    // }
+    // TODO: implement tag normalization
+    // let s: String = normalize_tag(svc);
+    // if s == "" {
+    //     return (fallbackService(lang), errors::NormalizeErrors::ErrorInvalid)
+    // }
+    // return (s, err)
+    // (svc, None)
+// }
+
+// normalize_name normalizes a span name and returns an error describing the reason
+// (if any) why the name was modified.
+pub fn normalize_name(name: String) -> (String, Option<errors::NormalizeErrors>) {
```

Comment:
I think you can use the built-in feature Result of rust instead of returning a tuple?

Reply:
I think the problem is this function has a case where the given name is over the max length, and this function needs to return the truncated name and an error indicating that the given name was too long

Reply:
I ended up refactoring these two functions to use Result<>